examples/train-text-from-scratch/train-text-from-scratch.cpp

```diff
@@ -1988,7 +1962,7 @@ void opt_callback(void * vdata, float * sched) {
     float min_sched = params->adam_min_alpha / params->adam_alpha;
     *sched = min_sched + *sched * (1.0f - min_sched);
 
-    int impr_plot = std::isnan(opt->loss_after) ? 0 : -(int)(1 + (opt->loss_before - opt->loss_after) * 10.0f + 0.5f);
+    int impr_plot = std::isnan(opt->loss_after) ? 0 : -std::lround(1 + (opt->loss_before - opt->loss_after) * 10.0f);
```

Comment:
This is unrelated to the PR, but I just saw this:

@xaedes When would we expect `nan`? Isn't it better to guarantee that we will never receive `nan` here?

Reply:
> When would we expect nan?

Actually only when messing up something with the optimizer or gradients *during development*.
When everything works as it should, then there should be no NANs.

When, during development, a NAN occurs the length of the "improvement plot bar" will be incredibly high, which will flood the console with single characters. The output before that could be useful in identifying the cause of the issue but will be lost.
To avoid flooding the console I added this ternary operator.